ecs-cluster-infrastructure-service-cloudfront.tf

```diff
@@ -0,0 +1,127 @@
+resource "random_password" "infrastructure_ecs_cluster_service_cloudfront_bypass_protection_secret" {
+  for_each = {
+    for k, v in local.infrastructure_ecs_cluster_services : k => v if v["enable_cloudfront"] == true && v["cloudfront_bypass_protection_enabled"] == true
+  }
+
+  length           = 32
+  special          = true
+  override_special = "123456890"
+}
+
+resource "aws_cloudfront_distribution" "infrastructure_ecs_cluster_service_cloudfront" {
+  for_each = {
+    for k, v in local.infrastructure_ecs_cluster_services : k => v if v["enable_cloudfront"] == true
+  }
+
+  enabled         = true
+  aliases         = ["${each.key}.${local.infrastructure_route53_domain}"]
+  is_ipv6_enabled = true
+  http_version    = "http2and3"
+  price_class     = "PriceClass_100"
+
+  viewer_certificate {
+    acm_certificate_arn            = local.enable_infrastructure_wildcard_certificate ? aws_acm_certificate_validation.infrastructure_wildcard_us_east_1[0].certificate_arn : null
+    cloudfront_default_certificate = local.enable_infrastructure_wildcard_certificate ? null : true
+    minimum_protocol_version       = local.enable_infrastructure_wildcard_certificate ? "TLSv1.2_2021" : null
```

Comment:
If `cloudfront_default_certificate` = `true` would you still need to include the `minimum_protocol_version`?

Reply:
The documentation is a bit conflicting/confusing:

> - `minimum_protocol_version`
> [...]
> Can only be set if `cloudfront_default_certificate = false`
> [...]
> If you have specified `cloudfront_default_certificate`, `TLSv1` must be specified.

In our case, `cloudfront_default_certificate` will never be specified as `false` (even though `false` is the default, it will only ever be specified as `true` or `null`) - which means `TLSv1` won't need to be specified, because we wont ever set `cloudfront_default_certificate` to `false`